dispatch.go

```diff
@@ -158,14 +178,8 @@ func (d *dispatch) RegisterProtocol(p Protocol) (err error) {
 	return err
 }
 
-func (d *dispatch) Reset() error {
-	d.callsMutex.Lock()
-	for k, v := range d.calls {
-		v.ch <- EofError{}
-		delete(d.calls, k)
-	}
-	d.callsMutex.Unlock()
-	return nil
+func (d *dispatch) Reset() {
+	close(d.stopCh)
```

Comment:
Maybe we need a better name than `Reset` here, which implies that it will start up again.  Maybe `close` instead?